src/DaedalusCompiler/Compilation/AssemblyBuilder.cs

```diff
@@ -337,15 +346,27 @@ public void execBlockEnd()
 
         public void assigmentStart(SymbolInstruction instruction)
         {
-            assigmentLeftSide = instruction;
+            assigmentLeftSide.Add(instruction);
+        }
+        
+        //TODO check if there are any possibilities of assigmentLeftSide longer than 2 instructions?
+        public void assigmentStart(SymbolInstruction instruction1, SymbolInstruction instruction2)
+        {
+            assigmentLeftSide.Add(instruction1);
+            assigmentLeftSide.Add(instruction2);
         }
 
         public void assigmentEnd(string assignmentOperator)
         {
-            var operationType = assigmentLeftSide.symbol.Type;
+            var operationType = assigmentLeftSide[assigmentLeftSide.Count - 1].symbol.Type;   //TODO check if there are any possibilities of assigmentLeftSide longer than 2 instructions?
             var assignmentInstruction = AssemblyBuilderHelpers.GetInstructionForOperator(assignmentOperator, true, operationType);
-            
-            addInstruction(assigmentLeftSide);
+
+            foreach (var instruction in assigmentLeftSide)
+            {
+                addInstruction(instruction);
```

Comment:
Please use `addInstructions` method instead of foreach

Reply:
Done